### PR TITLE
feat: add trait tags (colored boxes)

### PR DESCRIPTION
**Pull request by skaldarnar**
This adds _some_ colored boxes for traits and rarity.

The colors should be matching those of the books and https://pf2easy.com/.

This likely needs some adjustments before merging, for instance:
- reduce height of the tag row?
- should the tags be rendered with `tikz`, `colorbox`, some other package?
- move tags closer to card title

image

And to show the colors for "Rare" and "Uncommon" items:

image

### CODE REVIEW

File: cards.tex [hunk 1]
```diff
@@ -46,6 +46,22 @@
...
+}
+
+\definecolor{tagsBg}{RGB}{217, 196, 132}
+\definecolor{uncommonBg}{RGB}{152, 81, 61}
+\definecolor{rareBg}{RGB}{0, 38, 100}
+\definecolor{traitBg}{RGB}{94, 0, 0}
+
+\newcommand{\Trait}[1]{\Tag{traitBg}{#1}}
+\newcommand{\Rare}[0]{\Tag{rareBg}{Rare}}
+\newcommand{\Uncommon}[0]{\Tag{uncommonBg}{Uncommon}}
```
Comment (skaldarnar) [inline cards.tex:62]: Is this okay if no arguments are required? Or would one do this differently?

File: cards.tex [hunk 2]
```diff
@@ -73,7 +89,7 @@ \section{Frightened}
 
 \CheckFormula{strength, profic., item, status, circum.}
 
-\(\operatorname{1d4} + \FormulaVariable{strength}{\phantom{1}} \DamageType{Bludgeoning}\)
+\(\operatorname{1d4} + \FormulaVariable{strength}{\phantom{1}} \hfill \DamageType{Bludgeoning}\)
```
Comment (skaldarnar) [inline cards.tex:92]: Unrelated change.
```suggestion
\(\operatorname{1d4} + \FormulaVariable{strength}{\phantom{1}} \DamageType{Bludgeoning}\)
```

File: cards.tex [hunk 3]
```diff
@@ -135,6 +155,8 @@ \section{Frightened}
 \foreach[evaluate=\level as \bonus using int(\level*8)] \level in {1, ..., 10} {
 \Cantrip{\level}{Ray of Frost}
 
+\Traits{\Trait{Attack} \Trait{Cantrip} \Trait{Cold} \Trait{Evocation}}
```
Comment (Toxaris) [inline cards.tex:159]: "attack" and "cantrip" are already somewhere else on the card, do we want to repeat it here?

My thinking was to put an explicit "Cast" action on each spell card, and split the traits somehow cleverly between the spell itself and the action to cast it.

There is some complicated stuff going on with spell traits though. For example, if a spell needs [somatic](https://2e.aonprd.com/Rules.aspx?ID=283) components, the action to cast it gains the [manipulate](https://2e.aonprd.com/Traits.aspx?ID=104) trait. So should we put the words "somatic", "manipulate", or both on the card?

File: cards.tex (hunk 1 above)
Comment (Toxaris) [inline cards.tex:62]: `Rare` and `Uncommon` are helpful when looking in the books (because players need to ask the GM whether they can get these in a particular shop). But are they important on the item card when the player already acquired it?

File: cards.tex [hunk 4]
```diff
@@ -46,6 +46,22 @@
...
+\newcommand{\Tag}[2]{%
+    \tikz[baseline]{%
+        \node[anchor=base, text=white, fill=#1, font=\sffamily, text depth=.5mm] {#2};
+    }%
+}
+
+\definecolor{tagsBg}{RGB}{217, 196, 132}
+\definecolor{uncommonBg}{RGB}{152, 81, 61}
+\definecolor{rareBg}{RGB}{0, 38, 100}
+\definecolor{traitBg}{RGB}{94, 0, 0}
```
Comment (Toxaris) [inline cards.tex:58]: I understand that using the same colors as in the books helps with parsing the information. I'm a bit conflicted though because I don't actually like the colors ...

File: cards.tex (hunk 1 above)
Comment (Toxaris) [reply cards.tex:62]: 1. I think 0 is the default, so you can do `\newcommand{\Rare}{\Tag{rareBg}{Rare}}`
2. I think a nicer interface would be just `\tags{Rare, Uncommon, Magical, Extremely Cool, Overpowered}` and the `\tags` macro would know how to color which tag. That is a bit beyond my current LaTeX skills though :)

File: cards.tex [hunk 5]
```diff
@@ -46,6 +46,22 @@
 \newcommand{\Reference}[2]{{\small \color{gray} \engschrift #1 #2}}
 \newcommand{\DamageType}[1]{\operatorname{\mbox{#1}}}
 
+\newcommand{\Tag}[2]{%
+    \tikz[baseline]{%
+        \node[anchor=base, text=white, fill=#1, font=\sffamily, text depth=.5mm] {#2};
```
Comment (Toxaris) [inline cards.tex:51]: Should we use `\engschrift` or `\mittelschrift` here to stay with the look of a German traffic sign?

**Review by Toxaris — APPROVED** [review]

File: cards.tex [hunk 6]
```diff
@@ -110,6 +128,8 @@ \section{Frightened}
 \foreach[evaluate=\level as \bonus using int(\level*8)] \level in {1, ..., 10} {
 \Spell{\level}{Heal}
 
+\Traits{\Trait{Healing} \Trait{Necromancy} \Trait{Positive}}
+
```
Comment (Toxaris) [inline cards.tex]: There is not enough space on the "Heal" card for this, so it overflows into a second page :(

**Review by Toxaris — CHANGES_REQUESTED** [review]

File: cards.tex (hunk 3 above)
Comment (skaldarnar) [reply cards.tex:159]: For most things we try to be more explicity and include the rule, instead of relying on the player to remember the rule. So, I think spells with _somatic_ component should also have the _manipulate_ trait tag on the card. 

Having one tag row is a good start, we can easily split / distribute these tags over multiple places (like individual spells or attacks). Definitely needs more formatting, though 🙈

File: cards.tex (hunk 1 above)
Comment (skaldarnar) [reply cards.tex:62]: Do they affect costs or difficulty for _repairing_ the item? 

Is the information helpful to balance the game (for players and / or the GM) by knowing how many items from which category the party currently has?

Also, the cards might be used as a "shop front", and this makes it easier to draft a deck with say 1 rare, 3 uncommon, and _n_ common items...

File: cards.tex (hunk 4 above)
Comment (skaldarnar) [reply cards.tex:58]: Yes, the color scheme is ... meh. 

I wanted to start somewhere, and until I have a way better understanding of whether the traits can be grouped or categorized in a meaningful way this is as good as everything else for now. 

We can easily change that later as well.

File: cards.tex (hunk 6 above)
Comment (skaldarnar) [reply cards.tex]: Yes, we need to do something about the layout and unused whitespace. 

> This likely needs some adjustments before merging, for instance:
>
> - reduce height of the tag row?
> - move tags closer to card title

Btw, the **Steel Shield** card overflows already without tags.

File: cards.tex (hunk 1 above)
Comment (Toxaris) [reply cards.tex:62]: Ok, convinced :)
